File: Module10/PythonAndCSharpDemosForVS2015/MyCSharpApplication/Persons.cs
class Person
{
    //--Fields--
    static int __Counter = 0; 
    string __FirstName;

    //--Constructor--
    public Person(string FirstName = "")
    {
        //Attributes
        this.__FirstName = FirstName; // Private Attribute = Field in C#
        Person.__SetObjectCount(); // Private Static Method
    }

    //--Properties--
    //FirstName    
    public string FirstName
    {
        get
        { return __FirstName; }
        set
        { __FirstName = value; }
    }

    public override string ToString()
    {
        return this.FirstName;
    }

    static int GetObjectCount()
    { // You do not need the self keyword
        return Person.__Counter;
    }

    static void __SetObjectCount()
    { // This is a private and static method
        Person.__Counter += 1;
    }
}//--End of class Person--

File: Module10/PythonAndCSharpDemosForVS2015/MyPythonApplication/Persons.py
#--- Make the class ---
class Person(object):
    """ Base Class for Personal data """
    
    #--Fields--
    __Counter = 0 

    #--Constructor--
    def __init__(self, FirstName = ""):
        #Attributes
        self.__FirstName = FirstName # Private Attribute
        Person.__SetObjectCount() # Private Method
    
    #--Properties--
    #FirstName    
    @property #getter(accessor) 
    def FirstName(self):
        return self.__FirstName
    
    @FirstName.setter #(mutator)
    def FirstName(self, Value):
        self.__FirstName = Value
  
    #--Methods--
    def ToString(self):
        """Explictly returns field data"""
        return self.FirstName
    
    def __str__(self):
        """Implictly returns field data"""        
        return self.FirstName
    
    @staticmethod
    def GetObjectCount(): # You do not need the self keyword
        return Person.__Counter
    
    @staticmethod
    def __SetObjectCount(): # This is a private and static method
        Person.__Counter += 1 
#--End of class Person--